toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/TraceState.kt

```diff
@@ -468,6 +490,20 @@ public class TraceState(
         _currentTraceName.value = name
     }
 
+    internal fun selectNextCompletedTrace() {
+        val selectedTraceRunIndex = completedTraces.indexOf(_selectedTraceRun.value)
+        if (selectedTraceRunIndex + 1 < _completedTraces.size) {
+            _selectedCompletedTraceIndex = selectedTraceRunIndex + 1
+        }
+    }
+
+    internal fun selectPreviousCompletedTrace() {
+        val selectedTraceRunIndex = completedTraces.indexOf(_selectedTraceRun.value)
+        if (selectedTraceRunIndex - 1 >= 0) {
+            _selectedCompletedTraceIndex = selectedTraceRunIndex - 1
+        }
+    }
```

Comment:
This could also be simplified by using the current index:
```suggestion
    internal fun selectPreviousCompletedTrace() {
        if (currentTraceResultIndex.value > 0) {
            currentTraceResultIndex.value -= 1
        }
    }
```